### ISSUE TITLE
Add Yeast

### ISSUE THREAD

**Issue by IceDragon200** [GrowthcraftCE/Growthcraft-1.7#108]
[Forum Yeast Post](http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/wip-mods/2505072-growthcraft-community-edition-proper-1-7-10?comment=93)

Took time out to check the forum threads, and saw this comment

I know nothing about yeast and how its made, or what sorcery is needed to produce it, but if you guys can explain it to me, I can make it happen
## Checklist
- [x] Add Yeast Items
- [x] Implement methods for obtaining aforementioned yeast items
- [ ] Add new booze fermentations/brews using new yeast
  
  **For active modules**
  - [x] Apples
  - [x] Bees
  - [x] Grapes
  - [x] Hops
  - [x] Rice
  
  **For indev modules**
  - [x] Dairy
  - [x] Nether

**Comment by Sunconure11** [comment]
Yeast is a naturally occurring fungus, and can be found pretty much everywhere.

https://en.m.wikipedia.org/wiki/Yeast

**Comment by Sunconure11** [comment]
https://en.m.wikipedia.org/wiki/Saccharomyces_cerevisiae#Brewing

**Comment by Sunconure11** [comment]
https://en.m.wikipedia.org/wiki/Yeast_in_winemaking

**Comment by Sunconure11** [comment]
https://en.m.wikipedia.org/wiki/Bottom_and_top_fermenting_yeast#Fermentation_methods

**Comment by IceDragon200** [comment]
@Sunconure11 Ah, thanks, I'll read later today (for now, I must rest)

**Comment by spannerman79** [comment]
I _think_ another mod uses Yeast. Don't quote me on this but I think its Pams HarvestCraft or another mod like that which adds a cr@p load of food stuffs and one of them is yeast. I am unsure if its ordDict'ed or not but its something to consider.

Also there was a 1.6.4 mod that used yeast - [Kitchens](http://minecraft.gamepedia.com/Mods/Kitchens) with the following process

> 1. Craft 1 fruit (apple or grape or 1 slice of dry yeast!) with 1 or 3 water bottle crafts into 1 or 3 fruit water.
> 2. Place Fruit Water on the drying table. It will convert into Yeast Water.
> 3. Brew 1 Wheat with 3 Water bottles in a Cooking Stand to get 3 bottles of Grain water.
> 4. Craft 1 Yeast water with 6 Grain water to get 6 bottles of mash (you get the empty bottle back from yeast water).
> 5. Craft 3 wheat with 3 mash to get 1 Yeast Cake (you get the empty bottles back)
> 6. Craft 1 Yeast Cake with a knife to get 18 Yeast Slices. You do get your knife back. If you install the Booze mod without Kitchens, then you still don't need a knife.
> 7. Dry Yeast Slices in the Drying Table to get Dry Yeast.

**Comment by Ar97x** [comment]
This is getting interesting. Would be a great addition.

**Comment by ghost** [comment]
yeast is commonly made out of grapes peels. believe it or not. so if you add yeast simply add grapes -> yeast recipe. but then again this is a game not real life. 

the recipe above is too complicated
i'd simply do something like grape + water bucket => yeast + empty bucket

yeast is used to make beer or brew wine or make bread. you could replace nether wart with yeast. this will allow you to get early "potions' aka growthcraft brews. normal potions are still better but growthcraft ones are easier to make

**Comment by spannerman79** [comment]
> yeast is commonly made out of grapes peels. believe it or not

Yeast is a microorganism that also be obtained via potato peelings too. Among other sources as well.

**Comment by IceDragon200** [comment]
So from what I've collected the following are possible methods (in their simplest form)
- Wheat + Water + Sugar > Yeast
- Potato + Water + Sugar > Yeast
- (Fruit) + Water > Yeast

Right?

**Comment by Ar97x** [comment]
We could have the yeast produced on the brew kettle, as the second output
of Ale, No hops and sugar on the second input. And we could work on some
new drinks to make easier to produce yeast.

**Comment by IceDragon200** [comment]
Looks like its time to rewrite parts of the cellar registry to include register residue items (its currently fixed to Bonemeal for.. EVERYTHING :D)

**Comment by ghost** [comment]
i second the brew kettle . fruit + sugar + water let it brew output is yeast instead of bonemeal.  or probably any plant would work fine. even grass . with fruits/crops/vegetables you get higher chance of yeast maybe i don't know. maybe yeast should be a %chance rather than you always get some which is probably 100% for grapes/apples/potatoes and 50% for everything else . or not . maybe is a bad idea. but brew kettle is best .

**Comment by Alatyami** [comment]
In actual brewing you can reuse the left over yeast for another batch.

**Comment by ghost** [comment]
so you could make it be fruits, vegetables or even yeast to make more yeast + sugar + water in brewing cauldron. if you use fruits you get 1 yeast but if you use yeast you get 3 yeast. 

this yeast system might be cooler cause you kinda have to manufacture it yourself rather than use netherwarts. i always felt netherwarts are a bit odd choice

**Comment by Sunconure11** [comment]
What about some forms of wild yeast that can only be obtained by fermenting in certain biomes? https://en.wikipedia.org/wiki/Yeast_in_winemaking#Wild_yeasts_and_natural_fermentation

**Comment by Sunconure11** [comment]
Also, I forgot to make the still model because of stuff going on, but, I'll work on it when I have some free time. However, modelling blocks is not my specialty.

**Comment by ghost** [comment]
loooking forward to this one spent an entire day loooking for a damn nether fortress an nothing. i jsut wanna get drunk ;_;

i got about 6 barrels full of young unfermented brews .

**Comment by vadis365** [comment]
If this is implemented - I have just added a 'cultivated mould' block in The Erebus which is used as a fuel for a composter block - but at a stretch, and if it seems fitting, I would like to oreDict it to match your yeast ;) We also have a 'wild' mould that grows much more slow which could be used instead (if the cultivated variety seems OP)

**Comment by ghost** [comment]
@vadis365  what is this?? growthcraft working with erebus <3 there's a few things that require mention bamboo and honey being the biggest. i patched so growthcraft honey comb filled -> erebus honey drip and  erebus honey bucket + bucket ->  youngmead.  also  they want to make bee box from all types of mod woods including erebus?  sorry that was offtopic. there is a growthcraft api you guys could add to erebus would probably make it easier. i deff use both of these mods and theres a lot of crossover.

**Comment by IceDragon200** [comment]
@vadis365, @trollworkout Since I'm refactoring the API, I wouldn't recommend touching it just yet, unless you want to register bees, booze and vine drops (which is what it does currently)

API aside, the oreDict name, ahhh, `materialYeast` I suppose

**Comment by ghost** [comment]
@IceDragon200 u changed the honey i was like why is my honey so dim...deprecated craft 4 ... ahhhh ahahaha

**Comment by IceDragon200** [comment]
@trollworkout https://github.com/Alatyami/Growthcraft-1.7/issues/123

Here's what happened

Also would anyone like to make the yeast texture?

**Comment by Sunconure11** [comment]
I suppose I can make one, based on various powder textures. Or would you like something else?

Maybe color can be based on the environment, if we create versions of yeast found in certain environments only?

**Comment by Sunconure11** [comment]
Also, I found something while looking at powder textures to base this off of.

http://www.minecraftforum.net/forums/minecraft-discussion/suggestions/72791-ghast-powder

Maybe this can fit somewhere in the nether module? Maybe it makes nether crops grow to maturity instantly, and is crafted with ectoplasm as well?

**Comment by IceDragon200** [comment]
@Sunconure11 That would be cool, I'm currently refactoring the fermentation, so you can use different items for different booze, instead of just the nether wart, glowstone, redstone trio

**Comment by Sunconure11** [comment]
Whipped this up in a few minutes.

https://www.dropbox.com/s/uq0non3m27c0eoq/defaultyeast_brewers.png?dl=1

**Comment by Sunconure11** [comment]
I wrote down some ideas regarding yeast, and I am whipping up textures for most of these (they are mainly just recolors, because yeast is microscopic, and these are all in powdered forms):

Forms of yeast:

Brewers yeast: The most common form of yeast. Can be found randomly by breaking any fruit. Effects on brewing: None

Netherwart: Not truly a yeast, but, can be used as such. Effects on brewing: Speeds up brewing speed, without the nasty effects of other forms of yeast.

Ethereal Yeast: Found by leaving barrels in magic biomes, or rarely, breaking fruit in magic biomes. Effects on brewing: Strengthens the status effects (excluding tipsy), allowing for hyper-extended brews.

Lager Yeast: Found by leaving barrels in cool biomes, or rarely, breaking fruit in cool biomes. Effects on brewing: Allows for the creation of beverages such as lager. https://en.wikipedia.org/wiki/Lager

Bayanus Yeast: Found by breaking grapes rarely. Can be used for the creation of fortified wine due to its tolerance of high alcohol levels. Saccharomyces bayanus, https://en.wikipedia.org/wiki/Yeast_in_winemaking

Origin Yeast: A form of yeast found commonly in mushroom biomes by breaking mycelium, it ferments very fast, but, also inflicts alcohol with many nasty effects, such as heavy nausea, and a near 100% chance of getting at least tipsy 2 from beverages made using such yeast. It can also appear in barrels left in such biomes.

Netherrash: Using netherrash in brewing is not advised, as it can survive the heat and higher alcohol levels. All beverages created with it are effectively posioned with the Netherrash Effect, which infects players and mobs.

Yeast will only appear in a barrel if there is some unfermented liquid in it.

**Comment by Ar97x** [comment]
That's great. But what's your reference for magic biomes? Will this be like
a crossover with Thaum/Mistcraft?

**Comment by Sunconure11** [comment]
Thaumcraft, Ars Magica 2, and Biomes O Plenty have magical biomes.

Also, I created the textures for all of the yeast variants.

https://www.dropbox.com/s/fxxzh9o6021w8hf/yeastforms.zip?dl=0

**Comment by Sunconure11** [comment]
Any biome registered as magical in the biome dictionary would be a requirement, basically.

**Comment by IceDragon200** [comment]
@Sunconure11, @Ar97x Whew, this took some time (smashed my head against the code a bit)
This is what I've managed to implement thus far
- [x] Brewers Yeast - Can be used to ferment booze (however it cannot be obtained by normal means yet)
- [x] Nether Wart - is 33% faster than Brewers Yeast (needs config option)
- [x] Ethereal Yeast - Just leave a barrel in any MAGIC biome with a bit of young booze and you'll have some yeast after a period (need to add a config option for this)
- [x] Origin Yeast - same as above, just in a MUSHROOM biome
- [x] Lager Yeast - same as above, just in a COLD biome
- [ ] Netherrash - Not added, because this isn't the grcnether branch...

I haven't created the corresponding potion effects, liquids and so forth yet

**tl;dr**
**Fair Warning**: **Config was butchered**, so don't blame me when something explodes and the roof falls on your head

d4e1745 [Latest Build](https://dl.dropboxusercontent.com/u/72618186/minecraft/mods/growthcraft/2.3.1/growthcraft-1.7.10-2.3.1-complete--d4e1745.jar)

**Comment by Sunconure11** [comment]
For production of more yeast without being in the required biome, or tedious farming, what about some kind of solution that can be placed in a barrel, and if any form of yeast is added, it will slowly produce more yeast (at a cost of a small portion of the solution over time, though more solution can be added), in a fashion similar to bees? Maybe the solution could be made by mixing wheat, fruit, sugar, and water.

**Comment by Ar97x** [comment]
I like the biomes idea. It forces you to make other bases and find required
biomes. The solution idea seems helpful, but it would be too cheap to get
in my opinion. Another thing I was thinking about was some sort of big
tank, for mass production of yeast, maybe built in some sort of barrel-ish
pattern.

**Comment by Sunconure11** [comment]
Maybe 4 barrels in a circular pattern, with 5 buckets could create the yeast barrel?

**Comment by Ar97x** [comment]
I was actually thinking about multiblock stuff, but we may discuss that.

**Comment by IceDragon200** [comment]
4 barrels + 5 buckets!? WHAT SORCERY IS THIS!? (I already have a hard time with iron with all these tech mods, don't make me cry anymore >:)

I'd just add a new block called the "Ferment Jar", you just drop some booze in it and it uses it up to make yeast.

Plus, you can slap it on your counter right next to your favourite slice of cake (no, its not a lie).

One bucket worth of booze should produce around 16 yeast, that should be enough right?

WDYT @Sunconure11 @Ar97x

**Comment by Ar97x** [comment]
I really would like to see a big multiblock structure, producing batches of
yeast when full on young booze, something like the iron tanks from
Railcraft but made mostly of wood. And as we are making the bee boxes for
all wood variants, we could make it this way too.

**Comment by ghost** [comment]
lol is kinda like magic but then again minecraft is kinda funny like that so having barrels in odd placements or in certain biomes or from certain types of wood change up the yeast type or change up how much of it you get it's not surprising. adds a whole new fun mini game like layer to the game. maybe different yeasts could be used instead of redstone, glowstone, and netherwarts. certain yeasts make potent or extended or fermented etc.

so if you use yeast A you get fermented potent, if you use yeast B you get fermented extended etc. i dunno.

**Comment by Sunconure11** [comment]
Ferment jar could work.

**Comment by Sunconure11** [comment]
I need to deal with some crap going on with a friend, but, in the mean time, I'll post some colors for various beverages using various forms of yeast:

Ethereal yeast beverages:

Amrita: If one attempts to create Malice Cider using Ethereal Yeast, Amrita will be created instead. All of the buffs on it are stronger, but, so are the debuffs. Color: 16730112

Ambrosia: A far more stronger variant of wine, it also inflicts a health bonus in addition to its usual effects. Color: 4403246

Divine Sake: A form of rather strong sake, it has stronger buffs than normal sake, and it inflicts a haste buff in addition to its usual effects. Color: 16775885

Silken Nectar: Created using apples, it gives the player a random positive buff, lasting for 10 minutes. Color: 16749824

I'll work on more later, I'm going to go offline.

**Comment by Sunconure11** [comment]
Lager yeast beverages:

Lager: Brewing ale with lager yeast will create lager. Color: 10451025

Gelid Booze: Brewing malice cider with lager yeast will create this. Player is given a large haste buff instead of a large strength buff. Color: 9038320

Bayanus yeast beverages:

Port: Brewing wine with bayanus yeast will create port wine, a fortified wine. Color: 5120038

Netherrash beverages:

Vile Slop: Created by fermenting malice fruit with netherrash, this substance will inflict a random negative debuff lasting for 10 minutes, and grant the player a small strength and absorption bonus. However, it is heavily intoxicating, and will kill the player after 3 sips. Color: 4139540

All other beverages created with it are poisoned with netherrash debuffs

Origin yeast beverages:

Same as default, but, twice as intoxicating, and have a chance of inflicting the opposite debuff.

Netherwart and Brewer's Yeast beverages:

Same as default

**Comment by IceDragon200** [comment]
Gentlemen, I what's the multiplier for hyper-extended potion effects

extended is `2.67` for reference

**Comment by Sunconure11** [comment]
Maybe hyper-extended effects could be double that of extended

**Comment by Ar97x** [comment]
Could be. Or could be the potent effect with extended time (8min)

**Comment by PitchBright** [comment]
I circumvented the 'Netherwart is not Yeast' problem several months ago by retexturing and renaming netherwart to yeast, in a resourcepack.

I love where you guys are taking this Yeast idea. You know me though, I'm going to ask that you make the Yeast types by biome configurable  XD

Do you want that request opened in a new issue so you have it as a possible-to-do?

**Comment by PitchBright** [comment]
Sorry I missed the boat on the texture help request. In the future, if you guys want something done, just ping me in the #growthcraft irc channel. I'm almost always around, and I can churn stuff out for you pretty quickly.

**Comment by IceDragon200** [comment]
@PitchBright [Yeast Generation](https://github.com/Alatyami/Growthcraft-1.7/wiki/User-API-%282.4.x%29#yeast-generation)

**Comment by PitchBright** [comment]
You're killin' it today. What an awesome thing to wake up to! All these wicked configgables!

With the Biomes on the Yeast, can I also enter them by ID# and/or actual name (ie. 'Silkglades', 'Mangrove')?

**Comment by IceDragon200** [comment]
@PitchBright The configs expect the registered name (the one they used for the GameRegistry)

For example all Growthcraft yeast are `grc.yeast`

You can find out the registered names by dumping the data using NEI

So the item schema for yeast would be:

``` json
{
  "mod_id": "Growthcraft|Cellar",
  "name": "grc.yeast",
  "meta": 0, 
  "amount": 1
}
```

This is as simple as it gets (without over complicating the code side of things), you get pretty much a 1:1 mapping with the growthcraft api internals

**Comment by PitchBright** [comment]
Actually, I meant the biomes.

**Comment by IceDragon200** [comment]
@PitchBright [BiomeDictionary.Type](https://github.com/MinecraftForge/MinecraftForge/blob/1.7.10/src/main/java/net/minecraftforge/common/BiomeDictionary.java#L15-L64) names.

Nothing else.

**Comment by PitchBright** [comment]
Okay, no worries. Those "type" names are kinda useless when it comes to handling 100+ biomes (Biomes O Plenty, etc)… because of the way biomes are classified, where in reality they're very different environmentally. (Poor example, but Desert is type DRY, so is Antarctica biome)

Unless of course the config was able to do operands for biome types. But that's a nightmare to figure out how to exclude biomes by type as a server-admin/modpack-maker, when configuring for that many biomes.

That said, I'll put in a Feature Request down the road for additional biome config options there, as a Low Priority type thing… should you guys decide to make the whole Yeast-Flavouring system a thing.

**Comment by IceDragon200** [comment]
@Sunconure11 Hope you didn't mind, I copied your yeast concepts to [Growthcraft Cellar - Yeast](https://github.com/GrowthcraftCE/Growthcraft-1.7/wiki/Growthcraft-Cellar---Yeast)

**Comment by Sunconure11** [comment]
Also, would it be possible to add a variant of Lager without hops?

**Comment by IceDragon200** [comment]
@Sunconure11 Draft something up in the wiki and I'll implement it?

**Comment by Sunconure11** [comment]
Done.

**Comment by IceDragon200** [comment]
@Sunconure11 I just realized, I mis-tagged the Lager, it isn't actually hopped, its made from regular Ale so.. its was okay in the first place!

Brew up some Ale and then brew it again with Lager Yeast and you should get Lager

**Comment by Sunconure11** [comment]
I'll check it out later, looks like the latest build of Harvestcraft is incompatible with hunger overhaul.

**Comment by PitchBright** [comment]
Noooooooo! Don't say that :(

**Comment by Sunconure11** [comment]
It got fixed now

**Comment by Sunconure11** [comment]
Whenever I try to ferment ale, I get a message saying I cannot ferment this.

**Comment by Sunconure11** [comment]
Also, are the recipes for the new beverages in yet? I'm trying to replicate them but I cannot seem to do it.

**Comment by Sunconure11** [comment]
Also, if I give myself yeast from NEI, and have an existing type of yeast from this mod in my inventory, it will just add itself to pre-existing yeast until it becomes a stack, only then will it give me other forms of yeast. You might want to look into that. I'll create a gif of this later.

**Comment by IceDragon200** [comment]
@Sunconure11 Fixed the yeast problem!
Recipes are in the recipes are in, most of them are done by fermenting either the "fermented", or its "potent" or "extended"

I've decided to branch Lager off into its own booze group (its still apart of the hops module however)

**Comment by Sunconure11** [comment]
Is the recipe for port in? I'm creating ambrosia ATM.

**Comment by Sunconure11** [comment]
Shoving Bayanus Yeast does not seem to do anything.

**Comment by Sunconure11** [comment]
I've tried potent, extended, and fermented wine for creating port.

**Comment by Sunconure11** [comment]
I'm having the same issues with lager.

**Comment by Sunconure11** [comment]
I am only able to get default and ethereal yeast beverages working.

**Comment by Sunconure11** [comment]
Maybe a bunch of these issues could be solved by having a lot of the new beverages being able to be created via fermenting young booze with the various other forms of yeast?

Also, since we do not have any variants of mead yet, I'll write a few up:

Braggot: Created by mixing young mead with young ale in a kettle, and then fermenting it, this beverage has the effects of both mead and ale, but, is more intoxicating. Color: 7412510

Hopped Mead: Basically what it says on the tin. Mead with hops.

**Comment by IceDragon200** [comment]
@Sunconure11 Try the new build I just uploaded 

Changes: https://github.com/GrowthcraftCE/Growthcraft-1.7/compare/134d553...3edba59

**Comment by Sunconure11** [comment]
What is the current recipe for lager in this update?

**Comment by Sunconure11** [comment]
Origin yeast is working now, but, no lager or bayanus yeast beverages are able to be made.

**Comment by IceDragon200** [comment]
@Sunconure11 ~~https://gist.github.com/IceDragon200/d158ff9b7b78e36c94dc~~ See the table link below, working on a table, you can watch it

**Comment by Sunconure11** [comment]
Alright, hopefully this will help me figure it out.

**Comment by IceDragon200** [comment]
@Sunconure11 Better yet, I should have done this in google docs, gimme a sec

**Comment by IceDragon200** [comment]
@Sunconure11 https://docs.google.com/spreadsheets/d/1ykz1iqWUkarnZC76FGyjKs2wExdOgLJ7XHt3haNOV30/edit?usp=sharing

**Comment by Sunconure11** [comment]
Looks like the recipe you mentioned does nothing for port wine.

**Comment by Sunconure11** [comment]
I'm trying to test all of the new forms of booze, and it still cannot be replicated.

**Comment by Sunconure11** [comment]
Also, do you think it would be possible to create some kind of NEI support for brewing recipes?

Also, the document has a typo. For grape wine, it should be pressed from grapes, not pressed from apples.

**Comment by IceDragon200** [comment]
@Sunconure11 Finished the Table, I'll look into NEI

The effects really need balancing in that thing

**Comment by Sunconure11** [comment]
Looks like port and lager are currently not able to be made at the moment. I'll wait for that though. Maybe I'm doing something wrong and don't realize it.

**Comment by IceDragon200** [comment]
@Sunconure11 I made Port and Lager in development, sure you grabbed the latest build?

**Comment by Sunconure11** [comment]
I can see the items, I just cannot make them. I have the latest build.

**Comment by Sunconure11** [comment]
Even when I follow various instructions.

**Comment by IceDragon200** [comment]
@Sunconure11 [pictures](https://www.dropbox.com/sh/s5l969bbmos3s55/AADbzHm2wuwZ4J9-avm9_j73a?dl=0) tell a better story

**Comment by Sunconure11** [comment]
I thought you meant using the barrels. Following those instructions worked fine.

**Comment by PitchBright** [comment]
Long thread, I didn't read all the way back, but incase it hasn't been mentioned yet… there's Z-fighting on the jars: http://imgur.com/S53DKIW

**Comment by ghost** [comment]
what do you do with ethereal yeast or bayanus yeast? doesn't seem to work on anything

**Comment by IceDragon200** [comment]
@trollworkout Ethereal is used for creating Hyper Extended and most of the "special" beverages (ones with the fancy names), Bayanus is with only Grape Wine currently for creating fortified wine (Port Wine) via brewing, @PitchBright is helping us iron out the [recipes](https://docs.google.com/spreadsheets/d/1ykz1iqWUkarnZC76FGyjKs2wExdOgLJ7XHt3haNOV30/edit#gid=1739283064&vpid=A1)

**Comment by PitchBright** [comment]
Z-fighting is fixed.

**Comment by ghost** [comment]
@IceDragon200  thanks guys i'm gonna try this out

my favourite drink of all time is extended mead in a water bag . the regen you get counters poison and wither and many other similar damage over time effects

**Comment by ghost** [comment]
I'm thinking you guys can probably close this cause yeast is added everything works as intended :+1:  Haha . So much better to not have to hunt for netherwarts.

**Comment by IceDragon200** [comment]
@trollworkout Hai!

**Comment by ghost** [comment]
@IceDragon200  arigatou gozaimasu :))

**Comment by Sunconure11** [comment]
While yeast has been (mostly) completed, would it be possible to make it so that if you put different types of yeast in a yeast jar, you can set it so only produces that type of yeast? An example would be bayanus yeast. If you put bayanus yeast in a jar with some fluid, that jar will only produce bayanus yeast until you clear the jar.

**Comment by PitchBright** [comment]
Bayanus yeast is a hybridization of lager and brewers yeast. Easy solution to making Bayanus yeast something that can be acquired would be to add a simple crafting recipe for it (lager yeast + baynus yeast).

**Comment by IceDragon200** [comment]
@Sunconure11 It does work like that currently, well sorta.
Yeast Generation is broken into 2 steps
bootstrap or initialize
and produce

When it 'bootstraps', it randomly picks a yeast type from the available yeast list, this list is generated from the biome tags, which are picked from the jar's current biome.
This is done when the jar has no yeast yet.

Now when it produces, all subsequent yeast types will be the same **IF** the conditions for generating it are correct (it is valid in the biome of choice).

I can loosen the produce step to generate the same yeast present in the slot.

EDIT:
And done so, check #130 for the latest builds

**Comment by Ar97x** [comment]
Do we have NEI integration for the yeast jar?

**Comment by IceDragon200** [comment]
@Ar97x no, no we do not, not that we have NEI anything

**Comment by Ar97x** [comment]
And what about Waila?

**Comment by IceDragon200** [comment]
@Ar97x For Waila It displays the item and fluid content, just like all other cellar objects

**Comment by Ar97x** [comment]
I started playing with the new yeast system today, and I want to know if the Roofed Forest biome counts as a mushroom biome, since it has those mushroom trees all over. And another question, are the biomes for each yeast kind configurable?

**Comment by PitchBright** [comment]
@Ar97x The yeast biomes are configurable by type (tag) for now. Yeast by biome name is on the to-do list. I'll whip up a proper Issue for it. 1 sec. In the meantime… here's your config guide:
https://github.com/GrowthcraftCE/Growthcraft-1.7/wiki/User-API-(2.5.x)

**Comment by PitchBright** [comment]
FR'd in #197

**Comment by darthvader45** [comment]
Latest release build does not have a config option for origin yeast.  How would I go about editing my current yeast.json file?  Also, as said on my own issue, how does one get a bonus applied to the bee box?

**Comment by IceDragon200** [comment]
@darthvader45 Why dredge up an old and **CLOSED** issue?
Please open a new issue next time.

If you are running the latest dev build, then the config SHOULD contain an entry for origin yeast, assuming you followed by directions and removed the old config, OR checked the default config and copied it over.

`.minecraft/config/growthcraft/cellar/yeast.json.default`

``` json
{
  "comment": "Default Yeast Config",
  "data": [
    {
      "comment": "Brewers yeast is the default yeast, which appears in all other biomes that are filled by the Lager or Ethereal",
      "item": {
        "comment": "Brewers Yeast",
        "mod_id": "Growthcraft|Cellar",
        "name": "grc.yeast",
        "amount": 1,
        "meta": 0
      },
      "biome_types": [
        "HOT",
        "SPARSE",
        "DENSE",
        "WET",
        "DRY",
        "SAVANNA",
        "CONIFEROUS",
        "JUNGLE",
        "SPOOKY",
        "DEAD",
        "LUSH",
        "NETHER",
        "END",
        "OCEAN",
        "RIVER",
        "WATER",
        "MESA",
        "FOREST",
        "PLAINS",
        "MOUNTAIN",
        "HILLS",
        "SWAMP",
        "SANDY",
        "SNOWY",
        "WASTELAND",
        "BEACH",
        "DESERT",
        "FROZEN"
      ],
      "biome_names": []
    },
    {
      "comment": "Ethereal yeast is found in MAGICAL biomes, because its special",
      "item": {
        "comment": "Origin Yeast",
        "mod_id": "Growthcraft|Cellar",
        "name": "grc.yeast",
        "amount": 1,
        "meta": 3
      },
      "biome_types": [
        "MAGICAL"
      ],
      "biome_names": []
    },
    {
      "comment": "Lager yeast is found in COLD biomes, think snow places!",
      "item": {
        "comment": "Lager Yeast",
        "mod_id": "Growthcraft|Cellar",
        "name": "grc.yeast",
        "amount": 1,
        "meta": 1
      },
      "biome_types": [
        "COLD"
      ],
      "biome_names": []
    },
    {
      "comment": "Origin yeast is found in MUSHROOM biomes.",
      "item": {
        "comment": "",
        "mod_id": "Growthcraft|Cellar",
        "name": "grc.yeast",
        "amount": 1,
        "meta": 4
      },
      "biome_types": [
        "MUSHROOM"
      ],
      "biome_names": []
    }
  ]
}
```

As for bee box bonuses, use orange or red dye on the bee box.
